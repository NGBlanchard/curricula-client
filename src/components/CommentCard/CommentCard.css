.comment-author{
  color: rgba(203, 80, 255, 0.735);
  font-weight: bold;
  font-size: 17px
}

.comment-content{
  font-size: 18px;
}