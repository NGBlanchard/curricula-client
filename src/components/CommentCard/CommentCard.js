import React, { Component } from "react";
import { NiceDate, Bull } from "../Utils/Utils";
import CourseApiService from "../../services/course-api-service";
import CourseContext from "../../context/CourseContext";
import "./CommentCard.css";

export default class CommentCard extends Component {
  state = {
    user: ""
  };
  static contextType = CourseContext;

  componentDidMount = () => {
    this.context.clearError();
    CourseApiService.getUserById(this.props.comment.user_id).then(res =>
      this.setState({
        user: res.user_name
      })
    );
  };

  render() {
    const { comment } = this.props;
    return (
      <>
        <li key={comment.id} className="comment-card">
          <p className="comment-author">
            {this.state.user}
            <Bull />
            <NiceDate date={comment.date_created} />
          </p>
          <p className="comment-content">{comment.content}</p>
        </li>
      </>
    );
  }
}
