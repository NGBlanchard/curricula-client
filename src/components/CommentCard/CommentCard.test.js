import CommentCard from './CommentCard'
import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'


configure({ adapter: new Adapter() });

describe('<CommentCard />', () => {
  it('Renders without crashing', () => {
    shallow(<CommentCard/>)
  })
})