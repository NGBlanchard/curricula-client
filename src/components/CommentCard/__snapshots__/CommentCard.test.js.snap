// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentCard /> renders a Comment by default 1`] = `
<CommentCard
  comment={
    Object {
      "id": 1,
      "user_id": 2,
    }
  }
  user={2}
>
  <li
    className="comment-card"
    key="1"
  >
    <p
      className="comment-author"
    >
      <Bull>
        <span
          className="Bull"
        >
           â€¢ 
        </span>
      </Bull>
      <NiceDate>
        Invalid Date
      </NiceDate>
    </p>
    <p
      className="comment-content"
    />
  </li>
</CommentCard>
`;
